src/services/ApiService.ts

```diff
@@ -185,58 +185,54 @@ export class ApiService {
                     if (error.instancePath) {
                         console.log(`Removing invalid preference ${error.instancePath}`);
                         // Trim the leading "." from the path
                         delete preferences[error.instancePath.substring(1)];
+
+                        this.clearPreferences([error.instancePath.substring(1)]);
+                        deletedKeys.push(error.instancePath.substring(1));
                     }
                 }
+
+                if (deletedKeys.length > 0) {
+                    // Show an alert to the user about the deleted preferences
+                    const appStore = AppStore.Instance;
+                    appStore.alertStore.showAlert(`Deleted invalid preferences: ${deletedKeys.join(", ")}`);
+                }
             }
         }
         return preferences;
     };
 
     private upgradePreferences = (preferences: any) => {
         // Upgrade to V2 if required
-        if (preferences?.version && preferences.version === 1) {
-            if (preferences.astColor || preferences.astColor === 0) {
-                switch (preferences.astColor) {
-                    case 0:
-                        preferences.astColor = "auto-black";
-                        break;
-                    case 1:
-                        preferences.astColor = "auto-white";
-                        break;
-                    case 2:
-                        preferences.astColor = "auto-red";
-                        break;
-                    case 3:
-                        preferences.astColor = "auto-forest";
-                        break;
-                    case 4:
-                        preferences.astColor = "auto-blue";
-                        break;
-                    case 5:
-                        preferences.astColor = "auto-turquoise";
-                        break;
-                    case 6:
-                        preferences.astColor = "auto-violet";
-                        break;
-                    case 7:
-                        preferences.astColor = "auto-gold";
-                        break;
-                    case 8:
-                        preferences.astColor = "auto-gray";
-                        break;
-                    default:
-                        preferences.astColor = "auto-blue";
-                        break;
+        if (preferences["version"] === 1) {
+            // Convert preferences[PreferenceKeys.WCS_OVERLAY_AST_COLOR] from a number in version 1 to a string in version 2
+            // default to "auto-blue" if the value is not in the AST_COLORS map
+            const astColors = ["auto-black", "auto-white", "auto-red", "auto-forest", "auto-blue", "auto-turquoise", "auto-violet", "auto-gold", "auto-gray"];
+            const astColorKey = PreferenceKeys.WCS_OVERLAY_AST_COLOR;
+            if (astColorKey in preferences || preferences[astColorKey] === 0) {
```

Comment:
You don't need the `|| preferences[astColorKey] === 0`; that was only necessary when `preferences[astColorKey]` was being evaluated for truthiness (because zero is false). `astColorKey in preferences` replaces both of these with one check.